rskj-core/src/main/java/org/ethereum/config/SystemProperties.java

```diff
@@ -705,4 +705,50 @@ private Optional<List<BtcECKey>> getGenesisFederationPublicKeys() {
     public long getGasEstimationCap() {
         return configFromFiles.getLong(PROPERTY_RPC_GAS_ESTIMATION_CAP);
     }
+
+    public DbKind databaseKind() {
+        return DbKind.ofName(configFromFiles.getString("keyvalue.datasource"));
+    }
+
+    public DbKind getDbKindValueFromDbKindLogFile() {
+        try {
+            File file = new File(databaseDir(), "dbKind.properties");
+            Properties props = new Properties();
+            if (file.exists() && file.canRead()) {
+                try (FileReader reader = new FileReader(file)) {
+                    props.load(reader);
+                }
+
+                return DbKind.ofName(props.getProperty("keyvalue.datasource"));
+            }
+
+            return null;
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    public void generatedDbKindLogFile() {
+        DbKind dbKind = this.databaseKind();
+
+        try {
+            File file = new File(databaseDir(), "dbKind.properties");
+            Properties props = new Properties();
+            if (file.canRead()) {
+                try (FileReader reader = new FileReader(file)) {
+                    props.load(reader);
+                }
+            } else {
+                props.setProperty("keyvalue.datasource", dbKind.name());
+                file.getParentFile().mkdirs();
+                try (FileWriter writer = new FileWriter(file)) {
+                    props.store(writer, "Generated dbKind. In order to follow selected db.");
+                    logger.info("New nodeID generated: {}", props.getProperty("nodeId"));
```

Comment:
do you think we could have one log message here instead of two?

Reply:
This was a typo